### PR TITLE
Feature/ACS-937 migrate swagger UI to v3

### PR DESCRIPTION
**Pull request by andrea-ligios**
The main goal of this PR is to migrate the REST API Explorer from the legacy Swagger UI 2 (which is EOL) to the new Swagger UI 3.

Other than that, it's also a good idea to upgrade all the libraries used and to replace the legacy Tomcat7 Maven Plugin with the latest Jetty Maven Plugin.

### CODE REVIEW

**Review by mauriziovitale — APPROVED** [review]

**Comment by mauriziovitale** [comment]
> Hi @andrea-ligios if you could add some description also for future memory thanks

Do you mean doc?

**Review by gavincornwell — CHANGES_REQUESTED** [review]
Awesome job @andrea-ligios!

I ran the API Explorer locally and had a quick look, on the whole it works great, however, I did find the following error when expanding the POST /search endpoint.

Screenshot 2021-01-05 at 09 56 38

**Comment by eromano** [comment]
> > Hi @andrea-ligios if you could add some description also for future memory thanks
> 
> Do you mean doc?

I mean here in the PR:
No description provided.

**Comment by andrea-ligios** [comment]
@gavincornwell it's likely a [Swagger UI 3 issue](https://stackoverflow.com/q/65584131/1654265), nothing we should be able to address:

image

If you agree and approve the PR, we can merge.

**Comment by gavincornwell** [comment]
@andrea-ligios Thanks for investigating and digging deeper, we have indeed had to live with bugs in the older version too.

However, this one is quite prominent :-( Is it actually still possible to use the "Try it Out" button despite the error? The response mentions circular references, have you done a quick check of our definition?

**Comment by andrea-ligios** [comment]
> @andrea-ligios Thanks for investigating and digging deeper, we have indeed had to live with bugs in the older version too.
> 
> However, this one is quite prominent :-( Is it actually still possible to use the "Try it Out" button despite the error? The response mentions circular references, have you done a quick check of our definition?

It is possible to use the "Try it Out" button, and I've not found any problem in the API definition.
The `pivot` element is just rendered differently than before (`null` instead of`{}`) 

Swagger UI 2:
image

Swagger UI 3:
image

In either case, it needs to get populated properly before being passed as a parameter, so I'm going to document this error in the API definition and then wait for your final review to merge the PR.

File: src/main/webapp/definitions/alfresco-search.yaml
Comment (gavincornwell) [inline src/main/webapp/definitions/alfresco-search.yaml:335]: This is great, but personally, I would put this right at the beginning of the description so it's the first thing the user reads.

**Review by gavincornwell — APPROVED** [review]
LGTM, presuming the warning about the error in the search API is moved to the start of the description.

**Pull request merged**
